backend/src/main/java/com/google/sps/FindMatchQuery.java

```diff
@@ -14,42 +14,35 @@
 
 package com.google.sps;
 
-import com.google.gson.Gson;
 import com.google.sps.data.Match;
 import com.google.sps.data.Participant;
-import java.time.Clock;
-import java.time.Duration;
-import java.time.Instant;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Date;
-import java.util.Collection;
-import java.util.Collections;
+import java.time.ZonedDateTime;
 import java.util.List;
-import java.util.stream.Collectors;
 
 /** Class used to find a match in a list of Participants with the most recently added Participant */
 public final class FindMatchQuery {
 
-  private Clock clock;
+  private ZonedDateTime dateTime;
   private static int MAX_DURATION_DIFF = 15; // maximum difference in duration to be compatible
-  private static int PADDING_TIME = 15; // extra padding time to ensure large enough meeting time block
-  
+  private static int PADDING_TIME =
+      15; // extra padding time to ensure large enough meeting time block
+
   /** Constructor */
   public FindMatchQuery() {
-    clock = new Clock();
+    this(ZonedDateTime.now());
   }
 
   /** Constructor with manually set date */
-  public FindMatchQuery(Clock clock) {
-    this.clock = clock;
+  public FindMatchQuery(ZonedDateTime dateTime) {
```

Comment:
Please add javadocs to this method. Specifically as a user of this class I'd want to know what the preconditions are for the provided date time.